src/runtime/hvx_128.ll

```diff
@@ -474,3 +474,72 @@ define weak_odr void @halide.hexagon.scatter.release(i8* %ptr) nounwind uwtable
   call void asm sideeffect "vmem($0 + #0):scatter_release\0A; v1 = vmem($0 + #0)\0A", "=*m,*m,~{v1}"(i8* %ptr, i8* %ptr)
   ret void
 }
+
+declare <64 x i32> @llvm.hexagon.V6.vrmpybusi.128B(<64 x i32>, i32, i32)
+declare <64 x i32> @llvm.hexagon.V6.vrmpybusi.acc.128B(<64 x i32>, <64 x i32>, i32, i32)
+declare <64 x i32> @llvm.hexagon.V6.vrmpyubi.128B(<64 x i32>, i32, i32)
+declare <64 x i32> @llvm.hexagon.V6.vrmpyubi.acc.128B(<64 x i32>, <64 x i32>, i32, i32)
+
+define weak_odr <64 x i32> @halide.hexagon.vrmpy_odd.vub.vub.w(<128 x i8> %low_v, <128 x i8> %high_v, i32 %const) nounwind uwtable readnone {
+  %low = bitcast <128 x i8> %low_v to <32 x i32>
+  %high = bitcast <128 x i8> %high_v to <32 x i32>
+  %dv = call <64 x i32> @llvm.hexagon.V6.vcombine.128B(<32 x i32> %high, <32 x i32> %low)
```

Comment:
Should the intrinsics in hvx_128.ll take 2x wider vectors, and should we do a Shuffle::make_concat in the compiler code instead of vcombine at runtime? Would that maybe result in simpler IR sometimes?

Reply:
I wasn't sure if Shuffle::make_concat will result in a vcombine. I'll try your suggestion.

Reply:
Note that if you do this, some of these wrappers would become trivial and could be defined in CodeGen_Hexagon.cpp instead of the runtime.

Reply:
The problem I was facing there was that the immediate operand cannot be passed as an arguement.